### PR TITLE
Add release process file for pulsar-client-go

### PR DESCRIPTION
**Pull request by wolfstudy**
Signed-off-by: xiaolong.ran <rxl@apache.org>

### Modifications

- Add release process file for pulsar-client-go

### CODE REVIEW

**Comment by wolfstudy** [comment]
@sijie @tuteng @jiazhai @merlimat PTAL

File: docs/release-process.md [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+$ git tag -u $USER@apache.org v0.X.0-candidate-1 -m 'Release v0.X.0-candidate-1'
+
+# Push both the branch and the tag to Github repo
+git push origin branch-0.X.0
+git push origin v0.X.0-candidate-1
+```
+
+3. Build and inspect an artifact.
+
+Generate a release candidate package.
```
Comment (merlimat) [inline docs/release-process.md]: What's the point of releasing a src package? All the users of this library are going to fetch the sources from the Github repo in any case. 

I think we should be fine by providing an "official" tag.

File: docs/release-process.md [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+If you have not installed Go, install it according to the [installation instruction](http://golang.org/doc/install).
+
+> NOTE: Since the `go mod` package management tool is used in this project, **Go 1.11 or higher** version is required.
+
+After the build, inspect the artifacts:
+
+Run the following command to verify the license headers in the source files:
+
+```
+go test license_test.go
```
Comment (merlimat) [inline docs/release-process.md]: All these tests are being automatically run for every PR. I don't see much value in repeating them in manual form.

File: docs/release-process.md [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+```
+
+Run the standalone Pulsar service and check that the Go client can produce and consume messages correctly:
+
+```
+$ cd examples/consumer
```
Comment (merlimat) [inline docs/release-process.md]: Same here, we're running a more extensive set of tests for each commit. I don't think these manual steps will be useful in catching potential problems.

File: docs/release-process.md (hunk 1 above)
Comment (wolfstudy) [reply docs/release-process.md]: Yes, agree with you. will remove the step.

**Comment by wolfstudy** [comment]
@merlimat PTAL again, thanks.

**Pull request merged**